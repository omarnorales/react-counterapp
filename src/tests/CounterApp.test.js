import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';

import CounterApp from '../CounterApp';


describe('Tests on CounterApp', () => {

    let wrapper = shallow(<CounterApp value = { 10 } />);

    beforeEach(() => {

        wrapper = shallow(<CounterApp value = { 10 } />);

    })

    test('should show <CounterApp /> correctamente', () => {

        expect(wrapper).toMatchSnapshot();

    });

    test('should show counter', () => {

        const textH2 = wrapper.find('h2').text().trim();
        console.log(textH2);

        expect(textH2).toBe(`10`);
    });

    test('should increase counter +1 at click', async () => {
        const btn1 = wrapper.find('button').at(0).simulate('click');
        const textH2 = wrapper.find('h2').text().trim();
        console.log('Counter val: ',textH2);
        expect(textH2).toBe('11');
        
    });

    test('should decrease counter -1 at click', async () => {
        const btn1 = wrapper.find('button').at(2).simulate('click');
        const textH2 = wrapper.find('h2').text().trim();
        console.log('Counter val: ',textH2);
        expect(textH2).toBe('9');
        
    });

    test('should place default value with btn reset', () => {

        let wrapper = shallow(<CounterApp value = { 105 } />);

        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(0).simulate('click');
        wrapper.find('button').at(1).simulate('click');
        
        const counterText = wrapper.find('h2').text().trim();
        expect(counterText).toBe('105');

    });
    
    
})
