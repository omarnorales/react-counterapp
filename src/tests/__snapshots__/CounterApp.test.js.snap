// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on CounterApp should show <CounterApp /> correctamente 1`] = `
<Fragment>
  <h1>
    Counter
  </h1>
  <h2>
    10
  </h2>
  <button
    onClick={[Function]}
  >
    +1
  </button>
  <button
    onClick={[Function]}
  >
    Reset
  </button>
  <button
    onClick={[Function]}
  >
    -1
  </button>
</Fragment>
`;
